commit 3f63a4b3264cd07be766be6e42b7bbce24a14fc8 (refs/changes/29/3729/4)
Author: Dave Eddy <dave@daveeddy.com>
Date:   2018-03-29T14:05:32-04:00 (1 year, 6 months ago)
    
    TRITON-280 vmapi vms.full.test should block on test VMs to be destroyed fully
    Reviewed by: Todd Whiteman <todd.whiteman@joyent.com>
    Approved by: Todd Whiteman <todd.whiteman@joyent.com>

diff --git a/test/vms.full.test.js b/test/vms.full.test.js
index a4534dc..a292f0f 100644
--- a/test/vms.full.test.js
+++ b/test/vms.full.test.js
@@ -184,6 +184,9 @@ function createTestVms(cb) {
                     return;
                 }
 
+                assert.object(body, 'body');
+                assert.uuid(body.job_uuid, 'body.job_uuid');
+
                 var job = '/jobs/' + body.job_uuid;
 
                 i++;
@@ -241,8 +244,22 @@ function destroyTestVms(cb) {
                         return;
                     }
 
-                    ret.destroyedVms++;
-                    next();
+                    assert.object(delBody, 'delBody');
+                    assert.uuid(delBody.job_uuid, 'delBody.job_uuid');
+
+                    var job = '/jobs/' + delBody.job_uuid;
+
+                    waitForValue(job, 'execution', 'succeeded', {
+                        client: client
+                    }, function (waitForValueErr) {
+                        if (waitForValueErr) {
+                            next(waitForValueErr);
+                            return;
+                        }
+
+                        ret.destroyedVms++;
+                        next();
+                    });
                 });
 
             }, function (delVmsErr) {
